### ISSUE TITLE
Asynchronous loading of data for a portion of a form

### ISSUE THREAD

**Issue by cca-developer** [ngx-formly/ngx-formly#2803]
I would like to be able to asynchronously load data for a portion of a form.  For example, I might have my form separated into tabs, and I don't want to load the data for each tab until the user opens that tab.  Or even within a tab, I'd like to have the form organized with Material cards.  For some cards, I'd like to be able to load the data for that card asynchronously.
I'm thinking that this would work somewhat like the asynchronous loading for select options, but I'm not quite sure how it would work.
Basically, I'd like to have a custom type as part of a larger form, but the model data for that custom type is retrieved asynchronously.  So when the user navigates to the form, a portion of the form would appear immediately, but some cards would display a loading indicator until the data for those cards has been retrieved.
Do you have any examples for how to do this, or can you give some pointers for the correct direction to take to achieve this?

**Comment by cca-developer** [comment]
I attempted to do this by making a custom template.  This is what it looks like:
```
<mat-card>
  <mat-card-title>{{ to.label }}</mat-card-title>
  <mat-card-content>
    <ng-container *ngIf="to.asyncData | async; else loading">
      <formly-field *ngFor="let field of field.fieldGroup" [field]="field"></formly-field>
    </ng-container>
  </mat-card-content>
</mat-card>

<ng-template #loading>
  Loading...
</ng-template>
```
Then the component is defined by:
```
export class DrawingRevisionMaterialFormComponent extends FieldType implements FormlyExtension {
  formControl!: FormControl;

  onPopulate(field: FormlyFieldConfig) {
    // skip if the field is already initialized
    if (field.fieldGroup) {
      return;
    }

    field.fieldGroup = [
      {
        key: 'exactMaterial',
        type: 'input',
        templateOptions: {
          label: 'Equal Or Better / Exact'
        }
      },
      {
        key: 'materialRaw',
        type: 'material',
        templateOptions: {
          label: 'Raw Material'
        }
      },
      {
        key: 'materialNotes',
        type: 'textarea',
        templateOptions: {
          label: 'Material Notes'
        }
      }
    ];
    if (field.templateOptions) {
      field.templateOptions.asyncData = this.drawingRevisionsService.getRevisionMaterial().pipe(
        // The line below does not work because field.model is readonly.
        tap((material) => field.model = material)
      );
    }
  }

  constructor(private drawingRevisionsService: DrawingRevisionsService) {
    super();
  }

}
```
This works as far as displaying a loading indicator, fetching the data, and having a form that looks like what I want.  The problem is that the overall form model never gets the updated values.  As shown in the comment, `field.model` is readonly.  How can I update the model after the data has been fetched?

**Comment by aitboudad** [comment]
```ts
field.formControl.setValue(material)
```

**Comment by aitboudad** [comment]
whenever you modify the field on the field, make sure to call `_buildForm`:

```ts
options._buildForm();
````
